airdrop-member-nfts/index.js

```diff
@@ -0,0 +1,109 @@
+const { ethers } = require("ethers");
+const { NFTStorage, File } = require("nft.storage");
+const { CeloProvider, CeloWallet } = require('@celo-tools/celo-ethers-wrapper');
+const fs = require("fs");
+const Axios = require('axios');
+const mime = require("mime");
+
+const WALLET_PK = process.env.WALLET_PK;
+const NFT_ADDRESS = process.env.NFT_ADDRESS;
+const PROVIDER_URL = process.env.PROVIDER_URL;
+const TOKEN = process.env.NFT_STORAGE_TOKEN;
+const SEASON = process.env.SEASON;
+
+async function downloadImage(url, filepath) {
+  const response = await Axios({
+      url,
+      method: 'GET',
+      responseType: 'stream'
+  });
+  return new Promise((resolve, reject) => {
+      response.data.pipe(fs.createWriteStream(filepath))
+          .on('error', reject)
+          .once('close', () => resolve(filepath)); 
+  });
+}
+
+async function fileFromPath(filePath) {
+  const content = await fs.promises.readFile(filePath)
+  const type = mime.getType(filePath)
+  return new File([content], filePath.substring(5, filePath.length), { type })
+}
+
+exports.handler = async (event) => {
+  try{
+    const rawdata = fs.readFileSync("./CommunityMember.json");
+    const abi = JSON.parse(rawdata);
+
+    const provider = new CeloProvider(PROVIDER_URL);
+    const owner = new CeloWallet(WALLET_PK, provider);
+    const contract = new ethers.Contract(
+      NFT_ADDRESS,
+      abi.abi,
+      provider
+    );
+
+    if (event.wallet_id.length > 0) {
+      const balance = await contract.balanceOf(event.wallet_id);
+
+      if (balance > 0) {
+        const tokenId = await contract.tokenOfOwnerByIndex(event.wallet_id, 0);
+        return {
+          statusCode: 400,
+          body: {
+            errorId: 2,
+            error: "User already has the NFT",
+            tokenId: tokenId.toString(),
+            tokenAddress: NFT_ADDRESS
+          }
+        }
+      }
+      const tx = await contract.connect(owner).airdrop([event.wallet_id]);
+      const receipt = await tx.wait();
+
+      const transferEvent = receipt.events?.find((e) => {
+        return e.event === "Transfer";
+      });
+
+      const tokenId = transferEvent.args.tokenId.toNumber();
```

Comment:
if the tx fails there will be no transferEvent. should we do something in that case, like returning an error saying that happened with a different errorId?

Reply:
That already happens, if the TX fails an error is thrown and we're inside a try catch block